x-pack/libbeat/management/fleet/manager.go

```diff
@@ -0,0 +1,279 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package fleet
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"sync"
+
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/common/reload"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/management"
+	"github.com/elastic/beats/x-pack/agent/pkg/core/plugin/server"
+	"github.com/elastic/beats/x-pack/libbeat/management/api"
+
+	xmanagement "github.com/elastic/beats/x-pack/libbeat/management"
+)
+
+// ConfigManager provides a functionality to retrieve config channel
+// using which manager is informed about config changes.
+type ConfigManager interface {
+	ConfigChan() chan<- map[string]interface{}
+}
+
+// Manager handles internal config updates. By retrieving
+// new configs from Kibana and applying them to the Beat.
+type Manager struct {
+	config    *Config
+	logger    *logp.Logger
+	beatUUID  uuid.UUID
+	done      chan struct{}
+	registry  *reload.Registry
+	wg        sync.WaitGroup
+	blacklist *xmanagement.ConfigBlacklist
+
+	configChan chan map[string]interface{}
+}
+
+// NewFleetManager returns a X-Pack Beats Fleet Management manager.
+func NewFleetManager(config *common.Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	c := defaultConfig()
+	if config.Enabled() {
+		if err := config.Unpack(&c); err != nil {
+			return nil, errors.Wrap(err, "parsing fleet management settings")
+		}
+	}
+	return NewFleetManagerWithConfig(c, registry, beatUUID)
+}
+
+// NewFleetManagerWithConfig returns a X-Pack Beats Fleet Management manager.
+func NewFleetManagerWithConfig(c *Config, registry *reload.Registry, beatUUID uuid.UUID) (management.ConfigManager, error) {
+	var blacklist *xmanagement.ConfigBlacklist
+
+	if c.Enabled && c.Mode == xmanagement.ModeFleet {
+		var err error
+
+		// Initialize configs blacklist
+		blacklist, err = xmanagement.NewConfigBlacklist(c.Blacklist)
+		if err != nil {
+			return nil, errors.Wrap(err, "wrong settings for configurations blacklist")
+		}
+	}
+
+	log := logp.NewLogger(management.DebugK)
+
+	m := &Manager{
+		config:     c,
+		blacklist:  blacklist,
+		logger:     log.Named("fleet"),
+		done:       make(chan struct{}),
+		beatUUID:   beatUUID,
+		registry:   registry,
+		configChan: make(chan map[string]interface{}),
+	}
+
+	go m.startGrpcServer()
+
+	return m, nil
```

Comment:
lets make a followup issue to merge your `rejectlist` changes here.

Reply:
yep i was wondering why it is still named like that